### ISSUE TITLE
NullCoalescingAssignmentPass pairs mismatched get/set accessors into invalid ??=

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1462]
## Bug

`NullCoalescingAssignmentPass` raises the get/set diamond `if (obj.get_P() is null) obj.set_P(fallback);` to `obj.P ??= fallback;`, pairing the getter and setter by `PropertyName` + declaring type + same receiver/index operands only. It never verifies that the two accessors actually form one C#-spellable property/indexer — i.e. the getter return type equals the setter value-parameter type, the index parameter ref-kinds match, and both are static or both instance. Mismatched accessors are raised to an invalid/misleading `??=`.

Same over-raise class as the #1356 queue, and a sibling of #1416/#1420 (accessor-target spellability).

## Evidence

`src/ILInspector.Decompiler/Pipeline/Passes/NullCoalescingAssignmentPass.cs:116-134`:

```csharp
if (statement.HasElse
    || NullTested(statement.Condition) is not LoadProperty getter
    || statement.Then.Children is not [.., StoreProperty store]
    || store.PropertyName != getter.PropertyName
    || !Equals(store.Accessor.DeclaringType, getter.Accessor.DeclaringType)
    || !SameReceiver(getter.Instance, store.Instance)
    || !PlaceIdentity.SameOperands(getter.IndexArguments, store.IndexArguments))
{
    return null;
}
```

There is no check that `getter.Accessor.ReturnType` equals `store.Accessor.ParameterTypes[^1]` (the setter value), nor that index parameter ref-kinds match, nor that both accessors share static/instance-ness. A user type whose `get_Item(K)` and `set_Item(K, V)` are unrelated methods (different element types, or a by-ref index on one side) is paired and rendered `obj[k] ??= v`, which has no consistent C# indexer spelling, while reporting `Full`.

## First fix shape

Before raising, require the getter/setter to be the same spellable accessor pair:

- `getter.Accessor.ReturnType` equals the setter value-parameter type;
- index parameter lists match by type and ref-kind;
- both static or both instance.

Decline (leave the diamond) otherwise.

## Done signal

- Mismatched get/set accessor fixture (e.g. `get_Item(K): string` + `set_Item(K, int)`) stays as the if/diamond or degrades honestly; no `??=` is produced.
- Real `Prop ??= x` / `this[k] ??= x` positives still raise.

**Comment by richlander** [comment]
Already resolved by #1445 ("Require compatible property accessors for `??=`", #1400), which merged at 16:58Z — before this issue was filed at 17:44Z citing the pre-#1445 name-only code. `CompatiblePropertyAccessors` enforces all three asks: getter return == setter value type, index parameter types match (by-ref params are `TypeRef.ByRef(T)`, so a by-ref index on one side is unequal and declines), and both-static-or-both-instance. The `get_Item(K):string` + `set_Item(K,int)` done-signal and the by-ref-index example both decline today. PR #1486 pins the indexer-specific done-signal with regression tests (the existing decline test only covered a non-indexer property). NullCoalescingAssignmentPassTests 26/26.
